apps/backend/src/modules/users/user.service.ts

```diff
@@ -1,28 +1,35 @@
+import { BaseEncrypt } from "~/libs/modules/encrypt/base-encrypt.module.js";
 import { type Service } from "~/libs/types/types.js";
 import { UserEntity } from "~/modules/users/user.entity.js";
 import { type UserRepository } from "~/modules/users/user.repository.js";
 
 import {
 	type UserGetAllResponseDto,
+	type UserGetOneResponseDto,
 	type UserSignUpRequestDto,
 	type UserSignUpResponseDto,
 } from "./libs/types/types.js";
 
 class UserService implements Service {
+	private encrypt: BaseEncrypt;
```

Comment:
```suggestion
	private encrypt: Encrypt;
```

I think you need Type Encrypt to use here - not Base Class
and BaseEncrypt should implements Encrypt

Reply:
I don't have Encrypt type yet. Would it be a good practice to have it in cases like that?

Reply:
Yes, I'd create Encrypt type and use `class BaseEncrypt implements Encrypt`. You can check #37